vllm/worker/worker.py

```diff
@@ -157,19 +158,44 @@ def determine_num_available_blocks(self) -> Tuple[int, int]:
         # profiled peak memory.
         torch.cuda.synchronize()
         free_gpu_memory, total_gpu_memory = torch.cuda.mem_get_info()
+
         # NOTE(woosuk): Here we assume that the other processes using the same
         # GPU did not change their memory usage during the profiling.
         peak_memory = self.init_gpu_memory - free_gpu_memory
         assert peak_memory > 0, (
             "Error in memory profiling. This happens when the GPU memory was "
             "not properly cleaned up before initializing the vLLM instance.")
 
+        # Determine GPU memory available for KV-cache
+        available_gpu_memory = int(total_gpu_memory *
+                                   self.cache_config.gpu_memory_utilization)
+
+        # Ensure we have at least "minimum" dynamic memory
+        # (to accommodate possible dynamic allocations)
+        gb = 1024 * 1024 * 1024
+        min_dynamic_memory = int(1.5 * gb)
```

Comment:
My concern is that this can be inaccurate. It is reported in https://github.com/vllm-project/vllm/issues/5517 that different GPUs may cost different memory in terms of cudagraph.